### PR TITLE
api: reorganize key API

### PR DESCRIPTION
**Pull request by kyrofa**
This PR continues the reorganization of #188 by extracting the key API into its own module.

Note that this introduces no behavioral changes, it's just moving code around.

### CODE REVIEW

File: sros2/sros2/api/_key.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
+# Copyright 2019-2020 Canonical Ltd
```
Comment (ruffsl) [inline sros2/sros2/api/_key.py:1]: Nit: I think @mikaelarguedas commented on this same copyright notice from last PR.

Comment (kyrofa) [reply sros2/sros2/api/_key.py:1]: This one should be legit; I wrote some of these functions and modified others.

File: sros2/sros2/api/_key.py [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+        raise FileNotFoundError(errno.ENOENT, os.strerror(errno.ENOENT), keystore_path)
+    if not os.path.isdir(contexts_path):
+        return True
+    for name in os.listdir(contexts_path):
+        if os.path.isdir(os.path.join(contexts_path, name)):
+            print(name)
+    return True
+
+
+def _is_key_name_valid(name):
```
Comment (artivis) [inline sros2/sros2/api/_key.py:113]: Given that `create_key` is public, maybe this is worth being public too?

Comment (mikaelarguedas) [reply sros2/sros2/api/_key.py:113]: My understanding from https://github.com/ros2/sros2/pull/188#issuecomment-609054839 is that all API not currently used outside the module is made non-public.
After all the refactor PRs are merged we'll do another pass to decide which additional ones to make public

Comment (artivis) [reply sros2/sros2/api/_key.py:113]: Alright sounds good.

**Pull request merged**